tests/TestCase/BootstrapTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Muffin\Webservice\Test\TestCase;
+
+use Cake\Controller\Controller;
+use Cake\Core\Plugin;
+use Cake\Datasource\ConnectionManager;
+use Cake\Datasource\FactoryLocator;
+use Cake\TestSuite\TestCase;
+use Muffin\Webservice\Connection;
+use Muffin\Webservice\Test\test_app\Model\Endpoint\TestEndpoint;
+
+class BootstrapTest extends TestCase
+{
+    /**
+     * Test that the plugins bootstrap is correctly registering the Endpoint
+     * repository type with the factory locator
+     *
+     * @return void
+     */
+    public function testLoadingEndpointWithLoadModel()
+    {
+        Plugin::load('Muffin/Webservice', [
+            'path' => ROOT . 'src',
+            'bootstrap' => true
+        ]);
+
+        $connection = new Connection([
+            'name' => 'test',
+            'service' => 'Test'
+        ]);
+        ConnectionManager::setConfig('test_app', $connection);
+
+        $controller = new Controller();
+        $endpoint = $controller->loadModel('Test', 'Endpoint');
+
+        $this->assertInstanceOf(TestEndpoint::class, $endpoint);
+        $this->assertEquals('Test', $endpoint->getAlias());
+    }
+
+    public function testFactoryLocatorAddition()
+    {
+        $expected = [\Muffin\Webservice\Model\EndpointRegistry::class, 'get'];
```

Comment:
It's just dawned on me reviewing this, that this test case relies on the previous one. Should the call to `Plugin::load` be moved to `setUp()` ?